pytorch_lightning/trainer/connectors/env_vars_connector.py

```diff
@@ -0,0 +1,42 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from functools import wraps
+from typing import Callable
+
+from pytorch_lightning.utilities.argparse_utils import parse_env_variables, get_init_arguments_and_types
+
+
+def overwrite_by_env_vars(fn: Callable) -> Callable:
+    """
+    Decorator for :class:`~pytorch_lightning.trainer.trainer.Trainer` methods for which
+    input arguments should be moved automatically to the correct device.
```

Comment:
hehe, I think I know from where you copied this code :)) 
this docstring needs to be updated :)